src/devices/sound/roland_sa.cpp

```diff
@@ -0,0 +1,430 @@
+// license:BSD-3-Clause
+// copyright-holders:giulioz
+
+/*
+    Emulator for the gate arrays found in the Roland CPU-B board of SA-synthesis digital pianos.
+    Reverse engineering done from silicon images.
+    - IC19 R06-0001 (Fujitsu MB60VH142)
+    - IC9  R06-0002 (Fujitsu MB60V141)
+    - IC8  R06-0003 (Fujitsu MB61V125)
+
+    The system is essentially a sample player, which can play 16 voices, each one with 10 sample parts.
+
+    IC19 controls the CPU->RAM interface and computes the volume envelopes.
+    The CPU can control the envelopes by specifying 8 bits of destination and 8 bits of speed (bit 7 is polarity).
+    Every time the envelope reaches a destination, an IRQ is raised and the CPU can read which voice/part triggered it.
+
+    IC9 is a phase accumulator, which can take 16 bit of playback speed value and generate an address for the wave rom.
+    The upper part of the wave rom is fixed, making the samples at most 2048 long.
+
+    IC8 sums the 160 parts together into a buffer with 8 samples, performing interpolation and volume control.
+    Volume control is done in exponential space to avoid multiplication. The samples are in exponential format.
+*/
+
+#include "emu.h"
+#include "roland_sa.h"
+
+DEFINE_DEVICE_TYPE(ROLAND_SA, roland_sa_device, "roland_sa", "Roland SA CPU-B Sound Generator")
+
+// LUT for the address speed
+uint32_t env_table[] = {
```

Comment:
Those are way too generic names to put as globals.  Please plonk them as static const class members.

Reply:
Whoops, good point. Let me know if this it better

Reply:
I put it as static constexpr in the cpp file now, hope that works, otherwise sorry for the misunderstanding